tools/index.ts

```diff
@@ -177,21 +177,47 @@ export default class LexicalModelCompiler {
     // https://help.keyman.com/developer/cloud/model_info/1.0
     //
 
-    model_info.id = model_id; 
-    model_info.name = model_info.name || kmpJsonData.info.name.description;
-    model_info.authorName = model_info.authorName || kmpJsonData.info.author.description;
-    model_info.authorEmail = model_info.authorEmail || kmpJsonData.info.author.url;
+    function set_model_metadata(field, expected, warn?: boolean) {
+      if(model_info[field] && model_info[field] !== expected) {
+        if(warn || typeof warn === 'undefined')
+          console.warn(`Warning: source ${modelInfoFileName} field ${field} value "${model_info[field]}" does not match "${expected}" found in source file metadata.`);
+      }
+      model_info[field] = model_info[field] || expected;
+    }
+
+    // Merge model info file -- some fields have "special" behaviours -- see below
+
+    set_model_metadata('id', model_id);
+    set_model_metadata('name', kmpJsonData.info.name.description);
+    set_model_metadata('authorName', kmpJsonData.info.author.description);
+
+    // we strip the mailto: from the .kps file for the .model_info
+    set_model_metadata('authorEmail', kmpJsonData.info.author.url.match(/^(mailto\:)?(.+)$/)[2], false);
+    
+    // TODO: languages is an array so more complicated to compare
```

Comment:
This map of maps is difficult to comprehend. Could you refactor it to make it simpler, or add a comment explain what its purpose is?

Reply:
I'm not sure which map of maps you are referring to -- the languages property or something else? Can you elaborate?

Reply:
Ah sorry, I got it now -- your comment was on the `// comment ` line and I confused myself more than I needed to!